Added snackbar, story book

elayabharath opened the pull request:
- Added snackbar UI element.
- Setup css with variables [not sure if this is a good idea].
- Setup a storybook to show example of the element.

_Upcoming next_
Integrate snackbar to all repos

marianamarasoiu reviewed (commented):
Hi @elayabharath this is really great, thanks! first round of comments, mostly very minor.

Really great idea to use CSS variables - I've been thinking we should use them for a while, so great to use this opportunity to introduce them.

Also great to have a "storybook" to test these components out. What do you think about adding a button to the storybook that shows the snackbar when clicked?

On another front, I think we should also treat the CSS for the components themselves in the same way we treat the dart files, so based on https://dart.dev/tools/pub/package-layout it seems that the guidelines suggest that we can have css files alongside dart files in `lib/`. Given that we expect to have a few other css files that don't have a dart counterpart, I'm thinking it may be a good idea to keep the css files in `lib/css` instead rather than directly under `lib/`? Then any other css files that are styling the storyboard itself can live under `web/`

elayabharath commented on lib/snack_bar.dart:
```diff
@@ -0,0 +1,41 @@
+    _contents.text = message;
+    snackbarElement.classes.remove('hidden');
+    snackbarElement.setAttribute(
+        'type', type.toString().replaceAll('SnackbarNotificationType.', ''));
```
Have set it to 120, along with other settings using .vscode > settings.json and committed in here

elayabharath commented:
Fixed all the issues 
- Renamed `snack_bar` to `snackbar`
- settings.json / notably adds the new line at the end of files, 120 line width for folding
- moved css to under `/lib`
- Removed the config option on the snackbar time on screen

Could you take a look again? Thanks! @marianamarasoiu

marianamarasoiu commented on .vscode/settings.json:
```diff
@@ -0,0 +1,17 @@
+    "dart.lineLength": 120,
+    "[dart]": {
+        "editor.rulers": [
+            120
```
I just had a chat with Dan and we were discussing 160 for the backend code - I'd be inclined to revisit our 120 and use the same number here as well? what do you think?
```suggestion
            160
```

marianamarasoiu commented on web/script.dart:
```diff
@@ -0,0 +1,14 @@
+  SnackbarView snackbarView = SnackbarView();
+  snackbarContainer.append(snackbarView.snackbarElement);
+  snackbarTrigger.onClick.listen((_) {
+    snackbarView.showSnackbar("Welcome to KatiKati UI library!", SnackbarNotificationType.success);
```
nit:
```suggestion
    snackbarView.showSnackbar("Welcome to the Katikati UI library!", SnackbarNotificationType.success);
```

marianamarasoiu reviewed (approved):
thanks EB, LGTM with just a couple of very small comments!
